Camera2SlowMotion/app/src/main/java/com/example/android/camera2/slowmo/MainActivity.java

```diff
@@ -104,35 +111,100 @@ private void analyze() {
     mediaMetadataRetriever.setDataSource(getApplicationContext(), fileUri);
     totalFrames =
         Integer.valueOf(mediaMetadataRetriever.extractMetadata(METADATA_KEY_VIDEO_FRAME_COUNT));
-    List<Bitmap> frameList =
-        mediaMetadataRetriever.getFramesAtIndex(0, Math.max(0, totalFrames - SAFE_FRAMES));
-
-    for (int i = 0; i < frameList.size(); i++) {
-      // analysis will be done here (forking in a separate thread)
-      imageHolder = InputImage.fromBitmap(frameList.get(i), 0);
-      final int finalI = i;
-      Task<Text> result =
-          recognizer
-              .process(imageHolder)
-              .addOnSuccessListener(
-                  new OnSuccessListener<Text>() {
-                    @Override
-                    public void onSuccess(Text visionText) {
-                      results.add(visionText.getText());
-                      analyseResultField.append(
-                          "\n\nText detected at index:" + finalI + " " + visionText.getText());
-                      Log.d(
-                          "Rokus logs",
-                          "Text detected at index:" + finalI + " " + visionText.getText());
-                    }
-                  })
-              .addOnFailureListener(
-                  new OnFailureListener() {
-                    @Override
-                    public void onFailure(@NonNull Exception e) {
-                      Log.d("Rokus Logs:", "Analyse failed with: " + e.getMessage());
-                    }
-                  });
+    frameList = mediaMetadataRetriever.getFramesAtIndex(0, Math.max(0, totalFrames - SAFE_FRAMES));
+
+    AsyncTaskServer asyncTaskServer = new AsyncTaskServer();
+    asyncTaskServer.execute();
+
+    AsyncTaskOCR asyncTaskOCR = new AsyncTaskOCR();
+    asyncTaskOCR.execute();
+  }
+
+  private class AsyncTaskServer extends AsyncTask<String, String, String> {
```

Comment:
I don't understand what this code is doing, can you please add a comment or choose more descriptive names?

It seems to be receiving a series of timestamps from the server, but I'm not clear on what these are timestamps of or what you intend to do with them.